src/views/token-swap/components/offers/offers.view.jsx

```diff
@@ -0,0 +1,129 @@
+import PropTypes from 'prop-types'
+
+import useOffersStyles from './offers.style'
+import Spinner from '../../../shared/spinner/spinner.view'
+import {
+  getTokenAmountInPreferredCurrency,
+  getFixedTokenAmount
+  , CurrencySymbol
+} from '../../../../utils/currencies'
+
+function Offers ({
+  quotes,
+  fiatExchangeRates,
+  preferredCurrency,
+  selectedTokens,
+  selectedLpId,
+  onGoToQuotes,
+  onOpenOfferSidenav
+}) {
+  const classes = useOffersStyles()
+  const isBestQuote = selectedLpId === 'best'
+  const isLoading = quotes.status === 'loading'
+  const successful = quotes.status === 'successful'
+  const failure = quotes.status === 'failure'
+
+  const [quote, setQuote] = React.useState({ rate: 0 })
+  const [fiatReward, setFiatReward] = React.useState(0)
+  const [reward, setReward] = React.useState(0)
+  const [timeUntilValid, setTimeUntilValid] = React.useState(30000)
+
+  React.useEffect(() => {
+    if (quotes.status === 'successful' && quotes.data.quotes) {
+      const selectedQuote = isBestQuote
+        ? quotes.data.quotes[0]
+        : quotes.data.quotes.find(q => q.lpId === selectedLpId)
+      setQuote(selectedQuote)
+
+      const rewardAmount = getFixedTokenAmount(
+        selectedQuote.lpInfo.rewards[0].amount,
+        selectedTokens.to.token.decimals
+      )
+      const fiatRewardAmount = getTokenAmountInPreferredCurrency(
+        rewardAmount,
+        selectedTokens.from.token.USD,
+        preferredCurrency,
+        fiatExchangeRates
+      )
+
+      setFiatReward(fiatRewardAmount)
+      setReward(rewardAmount)
+      setTimeUntilValid(30000 - 1000)
+      // TODO should be launched with validUntil in miliseconds from quotes instead 30000
+    }
+  }, [quotes])
+
+  React.useEffect(() => {
+    if (timeUntilValid <= 0) return
+    const timer = setTimeout(() => setTimeUntilValid(timeUntilValid - 1000), 1000)
+    return () => clearTimeout(timer)
+  }, [timeUntilValid])
+
+  const msToTime = (ms) => {
+    let s = ms / 1000
+    const secs = s % 60
+    s = (s - secs) / 60
+    const mins = s % 60
+
+    return `${mins.toString().padStart(2, '0')}:${secs.toFixed(0).toString().padStart(2, '0')}`
+  }
+
+  const renderBtnText = {
+    successful: `Swap ${msToTime(timeUntilValid)}`,
+    failure: 'Insufficient liquidity'
+  }
+
+  return (
+    <div className={classes.root}>
+      {isLoading &&
+        <div className={classes.loading}>
+          <p>Searching for the best offers</p>
+          <Spinner />
+        </div>}
+      {successful &&
+        <div className={classes.offerBox}>
+          <div className={classes.row}>
+            <div className={classes.quote}>
+              <p className={classes.quoteText}>
+                {isBestQuote ? 'Best q' : 'Q'}uote from {quote.lpInfo?.name}
+              </p>
+              <p className={classes.quoteRate}>
+                1 {selectedTokens.from?.token.symbol}
+                = {quote.rate.toFixed(6)} {selectedTokens.to?.token.symbol}
```

Comment:
We have a constant in `constants.js` file called `MAX_TOKEN_DECIMALS`, instead of hardcoding the `6`.